test-suite/TestSpec.hs

```diff
@@ -171,6 +175,51 @@ spec = do
+            it "detects zero width" $ visibleLength narrow `shouldBe` 9
+        describe "measureAlignment" $ do
+            prop "agrees for ascii strings" $ \(ASCIIString x) -> measureAlignment (=='e') (WideString x) `shouldBe` measureAlignment (=='e') x
+            it "detects double width" $ measureAlignment (=='㐁') wide `shouldBe` AlignInfo 2 (Just 2)
+            it "fails to detect" $ measureAlignment (=='a') wide `shouldBe` AlignInfo 6 Nothing
+            it "detects zero width after" $ measureAlignment (=='n') narrow `shouldBe` AlignInfo 3 (Just 5)
+            it "detects zero width before" $ measureAlignment (=='r') narrow `shouldBe` AlignInfo 7 (Just 1)
+        describe "dropLeft" $ do
+            prop "agrees for ascii strings" $ \(Small n) (ASCIIString x) -> buildCell (dropLeft n (WideString x)) `shouldBe` dropLeft n x
+            describe "on wide characters" $ do
+                it "drops 1 character of double width" $ dropLeft 2 wide `shouldBe` WideString "㐁㐂"
+                it "drops 2 characters of double width and adds a space" $ dropLeft 3 wide `shouldBe` WideString " 㐂"
+            describe "on narrow characters" $ do
+                it "drops combining characters with their previous" $ dropLeft 7 narrow `shouldBe` WideString "r!"
+        describe "dropRight" $ do
+            prop "agrees for ascii strings" $ \(Small n) (ASCIIString x) -> buildCell (dropRight n (WideString x)) `shouldBe` dropRight n x
+            describe "on wide characters" $ do
+                it "drops 1 character of double width" $ dropRight 2 wide `shouldBe` WideString "㐀㐁"
+                it "drops 2 characters of double width and adds a space" $ dropRight 3 wide `shouldBe` WideString "㐀 "
+            describe "on narrow characters" $ do
+                it "drops a combining character for free" $ dropRight 3 narrow `shouldBe` WideString "Bien s"
+                it "does not drop a combining character without their previous" $ dropRight 2 narrow `shouldBe` WideString "Bien sû"
+
+    describe "wide text" $ do
+        describe "buildCell" $ do
+            prop "gives the same result as wide string" $ \x -> buildCell (WideText $ T.pack x) `shouldBe` x
+        describe "visibleLength" $ do
+            prop "gives the same result as wide string" $ \x -> visibleLength (WideText $ T.pack x) `shouldBe` visibleLength (WideString x)
+        describe "measureAlignment" $ do
+            prop "gives the same result as wide string" $ \x -> measureAlignment (=='e') (WideText $ T.pack x) `shouldBe` measureAlignment (=='e') (WideString x)
+        describe "dropLeft" $ do
+            prop "gives the same result as wide string" $ \(Small n) x -> buildCell (dropLeft n . WideText $ T.pack x) `shouldBe` dropLeft n (WideString x)
+        describe "dropRight" $ do
+            prop "gives the same result as wide string" $ \(Small n) x -> buildCell (dropRight n . WideText $ T.pack x) `shouldBe` dropRight n (WideString x)
```

Comment:
Please use space after `==`.  This also looks like a pattern that should be abstracted:
```
    measureAlignmentAt c = measureAlignment (== c)
```